test/foundry/FleekERC721/FleekERC721.ap.t.sol

```diff
@@ -0,0 +1,142 @@
+        CuT.addAccessPoint(tokenId, accessPointName);
+        CuT.removeAccessPoint(accessPointName);
+
+        expectRevertWithInvalidAP();
+        CuT.getAccessPointJSON(accessPointName);
+    }
+
+    function test_cannotRemoveUnexistentAccessPoint() public {
+        expectRevertWithInvalidAP();
+        CuT.removeAccessPoint("accesspoint.com");
+    }
+
+    function test_cannotTwiceRemoveAccessPoint() public {
+        string memory accessPointName = "accesspoint.com";
+        CuT.addAccessPoint(tokenId, accessPointName);
+        CuT.removeAccessPoint(accessPointName);
+
+        expectRevertWithInvalidAP();
+        CuT.removeAccessPoint(accessPointName);
+    }
+
+    function test_isAccessPointNameVerified() public {
+        string memory accessPointName = "accesspoint.com";
+        CuT.addAccessPoint(tokenId, accessPointName);
+        assertFalse(CuT.isAccessPointNameVerified(accessPointName));
+        CuT.setAccessPointNameVerify(accessPointName, true);
+    }
+
+    function test_increateAccessPointScore() public {
+        string memory accessPointName = "accesspoint.com";
+        CuT.addAccessPoint(tokenId, accessPointName);
+        assertAccessPointJSON(accessPointName, "0", "0", "false", "false", deployer);
+
+        CuT.increaseAccessPointScore(accessPointName);
+        assertAccessPointJSON(accessPointName, "0", "1", "false", "false", deployer);
+
+        CuT.increaseAccessPointScore(accessPointName);
+        assertAccessPointJSON(accessPointName, "0", "2", "false", "false", deployer);
+    }
+
+    function test_cannotDecreaseAccessPointScoreToMinusOne() public {
+        string memory accessPointName = "accesspoint.com";
+        CuT.addAccessPoint(tokenId, accessPointName);
+
+        assertAccessPointJSON(accessPointName, "0", "0", "false", "false", deployer);
+        expectRevertWithMinimalScore();
+        CuT.decreaseAccessPointScore(accessPointName);
+    }
+
+    function test_decreaseAccessPointScore() public {
+        string memory accessPointName = "accesspoint.com";
+        CuT.addAccessPoint(tokenId, accessPointName);
+
+        assertAccessPointJSON(accessPointName, "0", "0", "false", "false", deployer);
+        CuT.increaseAccessPointScore(accessPointName);
+        assertAccessPointJSON(accessPointName, "0", "1", "false", "false", deployer);
+        CuT.decreaseAccessPointScore(accessPointName);
+        assertAccessPointJSON(accessPointName, "0", "0", "false", "false", deployer);
+    }
+
+    function test_appAccessPoints() public {
+        CuT.addAccessPoint(tokenId, "accesspoint1.com");
+        CuT.addAccessPoint(tokenId, "accesspoint2.com");
+        CuT.addAccessPoint(tokenId, "accesspoint3.com");
+
+        string[] memory accessPoints = CuT.appAccessPoints(tokenId);
+        assertEq(accessPoints[0], "accesspoint1.com");
+        assertEq(accessPoints[1], "accesspoint2.com");
+        assertEq(accessPoints[2], "accesspoint3.com");
```

Comment:
I suggest we add an assertion here for the length of `accessPoints` to confirm it only contains three values.